### DISCUSSION TITLE
48% Makefile in my flutter app

### DISCUSSION THREAD

**Discussion by Daivy03** [github-linguist/linguist#6527] [Classification] [answered]
I have tried repeatedly to remove makefile from detection, but without succeeding so far. I would like to figure out which files it is detecting as makefiles. 
Repo: https://github.com/Daivy03/demiprof_flutter_app/tree/main
My **.gitattribute** so far:
```
*.dart linguist-language=Dart
*.swift linguist-language=Swift
*.h linguist-language=C
*.cpp linguist-language=C++
#Makefile cmake
Makefile linguist-vendored
*.cmake linguist-vendored
*.stash linguist-vendored=true
*.debug linguist-vendored=true
*.release linguist-vendored=true
**/Makefile linguist-vendored=true
```

### Expected language:
dart swift c++

### Detected language:
makefile cmake

**Comment by lildude** [comment, marked as answer]
This will be due to this one file: 

```
➜ github-linguist demiprof_flutter_app/ --breakdown
48.53%  221206     Makefile
40.18%  183157     Dart
5.14%   23441      C++
4.14%   18851      CMake
0.59%   2674       Swift
0.56%   2546       HTML
0.31%   1425       C
0.30%   1350       Java
0.16%   750        CSS
0.04%   188        JavaScript
0.03%   137        Kotlin
0.02%   80         Batchfile
0.01%   38         Objective-C

[...]

Makefile:
build/app/intermediates/flutter/debug/flutter_build.d
[...]
```

It's been detected as Make because of the `.d` extention:

https://github.com/github-linguist/linguist/blob/916bd8f3df802fa98b0ea85539e67bd0b88ef158/lib/linguist/languages.yml#L3974-L4008

**Reply by Daivy03** [reply]
Thanks for help. very accurate answer 👍

**Comment by Daivy03** [comment]
Updated the .gitattribute like below but it doesnt work:

```
**/Makefile linguist-vendored=true
*.d linguist-vendored=true
build/app/intermediates/flutter/debug/** linguist-vendored=true
build/app/intermediates/flutter/debug/flutter_build.d linguist-vendored=true
```

Where am i wrong?

**Reply by lildude** [reply]
> Updated the `.gitattribute` like below but it doesnt work:

Your error is here -------------^ You're missing an `s` from the filename.

The second line is also sufficient (you can optionally drop the `=true` too) as you don't have a Makefile in your repo.

**Reply by Daivy03** [reply]
what a rookie mistake ahah, thanks again for ur time
